core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterUpscaleService.java

```diff
@@ -103,6 +103,14 @@ public void installServicesOnNewHosts(Long stackId, String hostGroupName) throws
         recipeEngine.executePostAmbariStartRecipes(stack, hostGroup.getRecipes());
         ClusterApi connector = clusterApiConnectors.getConnector(stack);
         List<String> upscaledHosts = connector.upscaleCluster(hostGroup, runningInstanceMetaDataSet);
+        if (repair) {
```

Comment:
Is this going to work if both master and idbroker are down? I think after we replace the master we will try to restart the services which is going to fail.

Reply:
Yes, it will fail because CM will not be able to start every service, but it won't cause any problem. It will spend some time here to restart the node for sure which can be avoided. I can try to find out somehow if it is the last node we are trying to recover. Another solution can be if we check every other node are fine and if yes, then we can do a restart.

Reply:
How much is the "spend some time" in minutes? Do we need to consider this for e2e tests?